src/NuGet.Core/NuGet.LibraryModel/LibraryRange.cs

```diff
@@ -134,7 +134,7 @@ public bool Equals(LibraryRange other)
             }
 
             return TypeConstraint == other.TypeConstraint
-                && string.Equals(Name, other.Name)
+                && string.Equals(Name, other.Name, StringComparison.OrdinalIgnoreCase)
                 && Equals(VersionRange, other.VersionRange);
```

Comment:
Not related to this PR, but this looks like it is using reference equals for the version range instead of checking if the ranges are equal. It's probably faster this way, just kind of unexpected.